src/api/catalogueItem.tsx

```diff
@@ -191,3 +194,56 @@ export const useEditCatalogueItem = (): UseMutationResult<
     }
   );
 };
+
+export const useMoveToCatalogueItem = (): UseMutationResult<
+  TransferState[],
+  AxiosError,
+  MoveToCatalogueItem
+> => {
+  const queryClient = useQueryClient();
+  return useMutation(async (moveToCatalogueItem: MoveToCatalogueItem) => {
+    const transferStates: TransferState[] = [];
+    let successfulIds: string[] = [];
+
+    const promises = moveToCatalogueItem.selectedItems.map(
+      async (item: EditCatalogueItem, index) => {
+        return editCatalogueItem({
+          id: item.id,
+          catalogue_category_id:
+            moveToCatalogueItem.targetCatalogueCategory?.id,
+        })
+          .then((result) => {
+            transferStates.push({
+              name: result.name ?? '',
```

Comment:
```suggestion
              name: result.name,
```
Since result is now of type CatalogueItem, which always has a name.